src/sentry/ingest/partitioning.py

```diff
@@ -0,0 +1,45 @@
+"""
+The number of logical partitions used to distinguish between where records
+should be stored. These do not require individual physical slices but allow
+for repartitioning with less code changes per physical change.
+"""
+from typing import Literal
+
+from django.conf import settings
+
+Sliceable = Literal["generic_metrics_sets", "generic_metrics_distributions"]
+
+
+def map_org_id_to_logical_partition(org_id: int) -> int:
+    """
+    Maps an org_id to a logical partition. Since SENTRY_SLICING_LOGICAL_PARTITION_COUNT is
+    fixed, an org id will always be mapped to the same logical partition.
+    """
+    return org_id % settings.SENTRY_SLICING_LOGICAL_PARTITION_COUNT
+
+
+def map_logical_partition_to_slice(sliceable: Sliceable, logical_partition: int) -> int:
+    """
+    Maps a logical partition to a slice.
+    """
+
+    assert is_sliced(sliceable), f"cannot retrieve slice of non-partitioned sliceable {sliceable}"
```

Comment:
nit: This assert seems redudant to me. It resolves to `assert sliceable in settings.SENTRY_SLICING_CONFIG`, which is also what the next line asserts.

Reply:
you're right, good catch